v2/spiffetls/tlsconfig/config.go

```diff
@@ -8,63 +8,122 @@ import (
 	"github.com/spiffe/go-spiffe/v2/svid/x509svid"
 )
 
+type verifyPeerCertificate = func(raw [][]byte, certs [][]*x509.Certificate) error
+
+func wrapVerifyPeerCertificate(wrapped, newVerifier verifyPeerCertificate) verifyPeerCertificate {
```

Comment:
nit: move this unexported function below the exported ones...

Reply:
We should pass the wrapped VerifyPeerCertificate the verified chains of trust we get back from our verification step instead of passing the empty slice we get from the TLS stack.